rasa/core/test.py

```diff
@@ -600,48 +642,42 @@ def _collect_action_executed_predictions(
     expected_action = expected_action_name or expected_action_text
 
     policy_entity_result = None
+    prev_action_unlikely_intent = False
 
     if circuit_breaker_tripped:
         prediction = PolicyPrediction([], policy_name=None)
         predicted_action = "circuit breaker tripped"
     else:
-        action, prediction = processor.predict_next_action(partial_tracker)
-        predicted_action = action.name()
-
-        policy_entity_result = _get_e2e_entity_evaluation_result(
-            processor, partial_tracker, prediction
+        predicted_action, prediction, policy_entity_result = _run_action_prediction(
+            processor, partial_tracker, expected_action
         )
 
-        if (
-            prediction.policy_name
-            and predicted_action != expected_action
-            and _form_might_have_been_rejected(
-                processor.domain, partial_tracker, predicted_action
+    predicted_action_unlikely_intent = predicted_action == ACTION_UNLIKELY_INTENT_NAME
+    if predicted_action_unlikely_intent and predicted_action != expected_action:
+        partial_tracker.update(
+            WronglyPredictedAction(
```

Comment:
Wouldn't this and line 681 result in two successive `WronglyPredictedAction`? Could be understanding it wrongly since I am not familiar with `action_executed_eval_store`.

Reply:
That's possible but the first `WronglyPredictedAction` will be filtered out so it works exactly as expected :)